packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Primitives/ConstructorInitializer.cs

```diff
@@ -10,6 +10,6 @@ namespace Microsoft.Generator.CSharp.Primitives
 {
     public sealed record ConstructorInitializer(bool IsBase, IReadOnlyList<ValueExpression> Arguments)
     {
-        public ConstructorInitializer(bool isBase, IEnumerable<ParameterProvider> arguments) : this(isBase, [.. arguments.Select(p => p.AsExpression)]) { }
+        public ConstructorInitializer(bool isBase, IEnumerable<ParameterProvider> arguments) : this(isBase, [.. arguments.Select(p => (VariableExpression)p)]) { }
```

Comment:
Better to use an extension method.

Reply:
We can't use extension method or implicit casting here because this is covariance due to array type of ParameterProvider, we need to cast it explicitly.

Reply:
updated